eigensync/src/client/document.rs

```diff
@@ -0,0 +1,156 @@
+//! Automerge document management for client
+
+use crate::types::{Result, ActorId};
+use automerge::{AutoCommit, ChangeHash};
+use serde_json::Value;
+use std::collections::HashMap;
+
+/// Manager for Automerge documents
+pub struct DocumentManager {
+    documents: HashMap<String, AutoCommit>,
+    actor_id: ActorId,
+}
+
+impl DocumentManager {
+    /// Create a new document manager
+    pub fn new(actor_id: ActorId) -> Self {
+        tracing::debug!("Creating document manager for actor {}", actor_id);
+        
+        Self {
+            documents: HashMap::new(),
+            actor_id,
+        }
+    }
+
+    /// Get or create a document for a swap
+    pub fn get_or_create_document(&mut self, document_id: &str) -> Result<&mut AutoCommit> {
+        tracing::debug!("Getting or creating document {}", document_id);
+        
+        if !self.documents.contains_key(document_id) {
+            // TODO: Try to load from cache first
+            let mut doc = AutoCommit::new();
+            doc.set_actor(self.actor_id.0.clone());
+            self.documents.insert(document_id.to_string(), doc);
+        }
+        
+        Ok(self.documents.get_mut(document_id).unwrap())
```

Comment:
for testing `unwrap` is ok, but when we merge this should be `.expect("<reason this can't fail">)`